### PR TITLE
Fixes #672: Add option groups

### LINKED ISSUE #672: Define the cmd line options into groups for the help output
We had discussed grouping command line options earlier in the development process sot that they were organized within groups in the help display.  However, click did not offer that capability (Note: that this exists in argparse). About 6 months ago a new click contrib (click-group-options) was added to the click contrib project and that package now appears to be stable so the capability to group the command line argumetns exists for us.

This click contrib library works by installing the click-option-group package and causes no other internal changes to pywbemcli beyond changing he decorartors on the option definitions to break them into groups.

The issue for us is what is a logical  set of groupings that aids our users.

### PR DESCRIPTION
**Pull request by KSchopmeyer**
This PR adds the capability for grouping options in the help display and adds groups the general display.

Waiting on pr #675 which significatnly changes the test of the general help output before we integrate this into that test.

However, the help organization and messages are ready for review.

NOTE: At this point, the click_option_group package only supports python 3.6, 3.7, and 3.8.  Since there is no easy way for us to have the general options definition completely different for different versions of python, I am going to create an issue and pr for that project to get support extended to python 3.4 and 2.7.

Marked waiting for now.

### CODE REVIEW

File: docs/changes.rst [hunk 1]
```diff
@@ -26,6 +26,10 @@ Released: not yet
 
 **Enhancements:**
 
+* Group the general options into logical groups in the help output. This creates
+  groups of commands (server, server configuration , general) (see issue
+  # 672)
+>>>>>>> Fixes issue #672- Organize cmd line options by group
```
Comment (andy-maier) [inline docs/changes.rst:32]: Remove this remainder from the merge conflict.

**Comment by andy-maier** [comment]
Not yet ready for review - needs to wait until PR #675 is merged, then rebased and tests adjusted.

File: docs/changes.rst (hunk 1 above)
Comment (KSchopmeyer) [reply docs/changes.rst:32]: DONE

**Comment by andy-maier** [comment]
Waiting for start of pywbemtools 0.9.0

**Comment by andy-maier** [comment]
pywbemtools 0.9 has started, so this PR can resume.

**Comment by andy-maier** [comment]
See comment on issue #672. Closing this PR as a result. We may reopen it once we drop support for Python 2,7, 3.4 and 3.5.

**Pull request closed without merging**